services/core/shared/core/compat/sdk_v5/transactions.js

```diff
@@ -13,50 +13,88 @@
+			skimmedTransaction.amount = tx.asset.amount;
+			skimmedTransaction.recipientId = tx.asset.recipientAddress || null;
+
+			// TODO: Update the below params after accounts index is implemented
+			skimmedTransaction.recipientPublicKey = null;
+			skimmedTransaction.senderId = null;
+			return skimmedTransaction;
+		});
+		return transactions;
+	});
+	let allTransactions = [];
+	txnMultiArray.forEach(transactions => allTransactions = allTransactions.concat(transactions));
+	const result = await transactionsDB.writeBatch(allTransactions);
+	return result;
+};
 
 const normalizeTransaction = tx => {
-	const availableLiskModules = getRegisteredModules();
 	const txModule = availableLiskModules
 		.filter(module => module.id === String(tx.moduleID).concat(':').concat(tx.assetID));
-	tx.id = tx.id.toString('hex');
+	tx = parseToJSONCompatObj(tx);
 	tx.moduleAssetId = txModule[0].id;
 	tx.moduleAssetName = txModule[0].name;
-	tx.fee = Number(tx.fee);
-	tx.nonce = Number(tx.nonce);
-	tx.senderPublicKey = tx.senderPublicKey.toString('hex');
-	tx.signatures = tx.signatures.map(signature => signature.toString('hex'));
-	tx.asset.amount = Number(tx.asset.amount);
-	tx.asset.recipientAddress = tx.asset.recipientAddress.toString('hex');
 	return tx;
 };
 
 const getTransactions = async params => {
+	const transactionsDB = await knex('transactions');
 	const transactions = {
 		data: [],
 		meta: {},
 	};
-
-	// TODO: Remove the check. Send empty response for non-ID based requests
-	if (params.id || params.ids) {
-		const response = await coreApi.getTransactions(params);
-		if (response.data) transactions.data = response.data.map(tx => normalizeTransaction(tx));
-		if (response.meta) transactions.meta = response.meta;
-
-		// TODO: Indexed transactions to blockId
-		// transactions.data = await BluebirdPromise.map(
-		// 	transactions.data,
-		// 	async transaction => {
-		// 		const txBlock = (await getBlocks({ id: transaction.id })).data[0];
-		// 		transaction.unixTimestamp = txBlock.timestamp;
-		// 		return transaction;
-		// 	},
-		// 	{ concurrency: transactions.data.length },
-		// );
+	if (params.fromTimestamp || params.toTimestamp) {
+		params.propBetween = {
+			property: 'timestamp',
+			from: Number(params.fromTimestamp) || 0,
+			to: Number(params.toTimestamp) || Math.floor(Date.now() / 1000),
+		};
 	}
+	const resultSet = await transactionsDB.find(params);
+	if (resultSet.length) params.ids = resultSet.map(row => row.id);
+	const response = await coreApi.getTransactions(params);
+	if (response.data) transactions.data = response.data.map(tx => normalizeTransaction(tx));
+	if (response.meta) transactions.meta = response.meta;
+
+	transactions.data = await BluebirdPromise.map(
+		transactions.data,
+		async transaction => {
+			resultSet.filter(tx => {
+				if (tx.id === transaction.id) {
+					transaction.unixTimestamp = tx.timestamp;
+					transaction.height = tx.height;
+					transaction.blockId = tx.blockId;
+				}
+				return tx;
+			});
```

Comment:
Please refactor the code. `Array.filter()` usage here does not seem to fit.

Reply:
Done!